### PR TITLE
#160845504 Pagination support for articles

### PR DESCRIPTION
**Pull request by reiosantos**
#### What does this PR do?
- Adds pagination support for articles

#### Description of Task to be completed?
- Add pagination to our articles to make article traversal easier.
 
image

#### What are the relevant pivotal tracker stories?
[#160845504](https://www.pivotaltracker.com/story/show/160845504)

### CODE REVIEW

File: src/containers/viewAllArticles.js [hunk 1]
```diff
@@ -1,50 +1,78 @@
...
 
 	render() {
 		return (
-			<div className={`row ${styles['card-row']}`}>
-				<div className={`col s12 m12 l12 ${styles['card-col']}`}>
-					{this.state.articles.map(article => <ViewCard key={article.slug} article={article} />)}
+			<div>
+				<div className={`row ${styles['card-row']}`}>
+					<div className={`col s12 m12 l12 ${styles['card-col']}`}>
+						{this.state.articles.results.map(article => <ViewCard key={article.slug} article={article} />)}
```
Comment (isheebo) [inline src/containers/viewAllArticles.js]: here

File: src/containers/viewAllArticles.js [hunk 2]
```diff
@@ -1,50 +1,78 @@
...
+					</div>
 				</div>
+
+				<Pagination pageCount={this.state.pageCount} handlePageClick={this.handlePageClick} />
```
Comment (isheebo) [inline src/containers/viewAllArticles.js]: here

File: src/containers/viewAllArticles.js [hunk 3]
```diff
@@ -1,50 +1,78 @@
...
 	componentDidMount() {
 		this.timer = setInterval(this.loadAllArticles, 10000);
+		this.timer2 = setInterval(this.loadViewAll, 1000);
 	}
 
 	// noinspection JSUnusedLocalSymbols
 	componentWillReceiveProps(nextProps, nextContext) {
-		this.setState({articles: nextProps.articles.results});
+		this.setState(prevState => ({
+			articles: nextProps.articles,
```
Comment (isheebo) [inline src/containers/viewAllArticles.js]: here

File: src/containers/viewAllArticles.js (hunk 1 above)
Comment (reiosantos) [reply src/containers/viewAllArticles.js]: This issue has been resolved.

File: src/containers/viewAllArticles.js (hunk 2 above)
Comment (reiosantos) [reply src/containers/viewAllArticles.js]: This too has been resolved.

File: src/containers/viewAllArticles.js (hunk 3 above)
Comment (reiosantos) [reply src/containers/viewAllArticles.js]: I have fixed this to use destructuring assignment.

**Review by huxaiphaer — DISMISSED** [review]

File: src/containers/pagination.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import ReactPaginate from 'react-paginate';
+import {Link} from 'react-router-dom';
+import React from 'react';
+import * as PropTypes from 'prop-types';
```
Comment (isheebo) [inline src/containers/pagination.js:4]: no need for the `* as` in this import

Comment (reiosantos) [reply src/containers/pagination.js:4]: This has been fixed @isheebo. I have removed the `*`

**Review by Estaer — APPROVED** [review]

**Review by huxaiphaer — APPROVED** [review]

**Review by DrKimpatrick — APPROVED** [review]

**Review by collinewait — APPROVED** [review]

**Pull request merged**